crates/hir-ty/src/infer/unify.rs

```diff
@@ -661,6 +700,38 @@ impl<'a> InferenceTable<'a> {
         }
     }
 
+    fn try_fulfill_obligation(
```

Comment:
I guess making `try_resolve_obligation` resolve the `solution` instead of bool kind of fixes the duplication as well as confusion. I'm not too sure where the `FIXME` comments of last 2 match arms should go, to the callee or stay here